### PR TITLE
fix(sdk): entity picker theming fixes

### LINKED ISSUE #50394: Make entity picker's placeholder, chevron and search picker color themeable in embedding sdk
Based on https://github.com/metabase/metabase/pull/50299, there are still a couple of visual theming issues:

- Placeholder color is always white on load
- The right chevron icon on the collection picker is always black
- The "Search Everywhere / Our Analytics" selector is not yet themed

CleanShot 2567-11-21 at 03 02 06@2x

CleanShot 2567-11-21 at 03 02 18@2x

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/50394

### Description

Adds additional theming fixes for the entity picker so it follows the SDK theming.

### How to verify

- Open Storybook on the SDK and click on the "Dark Theme Editor" stories

CleanShot 2567-11-25 at 18 45 08@2x

- There should be no noticeable change in the non-themed "Editor Only" story. Remember to _reload_ the page after switching between stories, as there is a bug caused by patching the `colors` object that caused the themed colors to persist across stories.

### Demo

Themed Chevron Color

Themed Segment Control

CleanShot 2567-11-25 at 18 40 32@2x

Themed Skeleton: [see private Loom](https://www.loom.com/share/ac01f71df753402abc4592577294f8f1?sid=05d4101c-7a17-4071-9b8c-3094d7a81394)

### CODE REVIEW

File: frontend/src/metabase/common/components/EntityPicker/components/ItemList/ItemList.tsx [hunk 1]
```diff
@@ -78,12 +78,22 @@ export const ItemList = <
         const isSelected = isSelectedItem(item, selectedItem);
         const icon = getEntityPickerIcon(item, isSelected && isCurrentLevel);
 
+        const folderChevronColor = isSelected
+          ? "var(--mb-color-text-white)"
+          : "var(--mb-color-text-primary)";
```
Comment (heypoom) [inline frontend/src/metabase/common/components/EntityPicker/components/ItemList/ItemList.tsx:83]: Allows the right chevron color of unselected items to be themed:

File: frontend/src/metabase/ui/components/feedback/Skeleton/Skeleton.styled.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const getSkeletonOverrides = (): MantineThemeOverride["components"] => {
         return {
           // We replace Mantine's pulsing animation with a shimmer animation
           root: {
-            backgroundColor: "rgba(245,248,250,1.0)",
+            backgroundColor: "var(--mb-color-background-disabled)",
```
Comment (heypoom) [inline frontend/src/metabase/ui/components/feedback/Skeleton/Skeleton.styled.tsx:19]: Allows [loading skeletons to be themed](https://www.loom.com/share/ac01f71df753402abc4592577294f8f1?sid=7c71ef7a-abf6-417c-bb0f-f9e6b08c2b96). The color of `background-disabled` is the closest possible to `rgba(245, 248, 250, 1.0)` and also makes semantic sense.

File: frontend/src/metabase/ui/components/inputs/SegmentedControl/SegmentedControl.styled.tsx [hunk 3]
```diff
@@ -51,10 +51,12 @@ export const getSegmentedControlOverrides =
...
           controlActive: {
-            backgroundColor: shouldAnimate ? theme.white : undefined,
+            backgroundColor: shouldAnimate
+              ? theme.fn.themeColor("bg-white")
+              : undefined,
           },
           indicator: {
-            backgroundColor: theme.white,
+            backgroundColor: theme.fn.themeColor("bg-white"),
           },
```
Comment (heypoom) [inline frontend/src/metabase/ui/components/inputs/SegmentedControl/SegmentedControl.styled.tsx:60]: Allows the segmented control to be themed:

CleanShot 2567-11-25 at 18 40 32@2x

**Review by deniskaber — APPROVED** [review]

File: frontend/src/metabase/common/components/EntityPicker/components/ItemList/ItemList.tsx (hunk 1 above)
Comment (heypoom) [reply frontend/src/metabase/common/components/EntityPicker/components/ItemList/ItemList.tsx]: Nice catch! Fixed in https://github.com/metabase/metabase/pull/50449/commits/6e4599e4c607de8d1eb57f3338966eed629880dc

**Review by WiNloSt — APPROVED** [review]

File: frontend/src/metabase/common/components/EntityPicker/components/ItemList/ItemList.tsx [hunk 4]
```diff
@@ -78,12 +78,22 @@ export const ItemList = <
         const isSelected = isSelectedItem(item, selectedItem);
         const icon = getEntityPickerIcon(item, isSelected && isCurrentLevel);
 
+        const folderChevronColor = isSelected
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/common/components/EntityPicker/components/ItemList/ItemList.tsx]: Thanks for the suggestion! Fixed in https://github.com/metabase/metabase/pull/50449/commits/1483e32af0a9c3eea6d6c8ef11d3f43606b88ebf

**Comment by heypoom** [comment]
@metabase-bot backport release-x.51.x

**Comment by heypoom** [comment]
@metabase-bot backport release-x.51.x

**Comment by heypoom** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**